src/test/java/io/cdap/plugin/sfmc/sink/DataExtensionClientTest.java

```diff
@@ -0,0 +1,201 @@
+        Schema.nullableOf(Schema.of(Schema.LogicalType.TIMESTAMP_MICROS))));
+    List<ETDataExtensionColumn> columnList = new ArrayList<>();
+    String dataExtensionKey = "DE";
+    PowerMockito.mockStatic(ETDataExtension.class);
+    DataExtensionClient dataExtensionClient = Mockito.spy(new DataExtensionClient(getClient(), dataExtensionKey));
+    Mockito.when(dataExtensionClient.getDataExtensionInfo()).thenReturn(null);
+    PowerMockito.when(ETDataExtension.retrieveColumns(Mockito.any(), Mockito.anyString())).thenReturn(columnList);
+    try {
+      dataExtensionClient.validateSchemaCompatibility(schema, collector);
+      collector.getOrThrowException();
+      Assert.fail("The columns are null");
+    } catch (ValidationException e) {
+      Assert.assertEquals("Errors were encountered during validation.", e.getMessage());
+    }
+  }
+
+  @Test
+  public void testValidateSchemaCompatibilityWNullSchemaField() throws ETSdkException {
+    MockFailureCollector collector = new MockFailureCollector();
+    Schema schema = Schema.recordOf("record",
+      Schema.Field.of("id", Schema.of(Schema.Type.LONG)),
+      Schema.Field.of("timestamp",
+        Schema.nullableOf(Schema.of(Schema.LogicalType.TIMESTAMP_MICROS))));
+    List<ETDataExtensionColumn> columnList = new ArrayList<>();
+    Collection<ETDataExtensionColumn> columns = new ArrayList<>();
+    ETDataExtensionColumn column = new ETDataExtensionColumn();
+    Date d = new Date(2022, 02, 9);
+    column.setCreatedDate(d);
+    column.setDefaultValue("default");
+    column.setName("dt");
+    columns.add(column);
+    String dataExtensionKey = "DE";
+    columnList.add(column);
+    column.setIsRequired(true);
+    PowerMockito.mockStatic(ETDataExtension.class);
+    DataExtensionClient dataExtensionClient = Mockito.spy(new DataExtensionClient(getClient(), dataExtensionKey));
+    PowerMockito.when(ETDataExtension.retrieveColumns(Mockito.any(), Mockito.anyString())).thenReturn(columnList);
+    try {
+      dataExtensionClient.validateSchemaCompatibility(schema, collector);
+      collector.getOrThrowException();
+      Assert.fail("Schema field is null");
+    } catch (ValidationException e) {
+      Assert.assertEquals("Errors were encountered during validation.", e.getMessage());
+    }
+  }
+
+  @Test
+  public void testScan() throws ETSdkException {
+    String dataExtensionKey = "DE";
+    List<ETDataExtensionRow> row = new ArrayList<>();
+    ETDataExtensionRow row1 = new ETDataExtensionRow();
+    row1.setId("id");
+    row.add(row1);
+    ETResponse<ETDataExtensionRow> response = new ETResponse<>();
+    response.setStatus(ETResult.Status.ERROR);
+    PowerMockito.mockStatic(ETDataExtension.class);
+    PowerMockito.when(ETDataExtension.select((ETClient) Mockito.any(), Mockito.anyString())).thenReturn(response);
+    DataExtensionClient dataExtensionClient = Mockito.spy(new DataExtensionClient(getClient(), dataExtensionKey));
+    dataExtensionClient.scan();
```

Comment:
what does scan do? Seems you're not validating anything

Reply:
Added the validation.